src/hooks/index.js

```diff
@@ -489,147 +578,183 @@ exports.mapPayload = (context) => {
-	if (context.type === 'before') {
-		let populate = context.params.query.$populate;
-		if (typeof (populate) === 'undefined') {
-			populate = ['courseId', 'courseGroupId'];
-		} else if (Array.isArray(populate) && !populate.includes('courseId')) {
-			populate.push('courseId');
-			populate.push('courseGroupId');
+exports.restrictToUsersOwnLessons = (context) =>
+	getUser(context).then((user) => {
+		if (testIfRoleNameExist(user, ['superhero', 'administrator'])) {
+			return context;
 		}
-		context.params.query.$populate = populate;
-	} else {
-		// after-hook
-		if (context.method === 'get' && (context.result || {})._id) {
-			let tempLesson = [context.result];
-			tempLesson = tempLesson.filter((lesson) => {
-				if ('courseGroupId' in lesson) {
-					return userIsInThatCourse(user, lesson.courseGroupId, false);
-				}
-				return userIsInThatCourse(user, lesson.courseId, true)
-					|| (context.params.query.shareToken || {}) === (lesson.shareToken || {});
-			});
-			if (tempLesson.length === 0) {
-				throw new Forbidden("You don't have access to that lesson.");
+		// before-hook
+		if (context.type === 'before') {
+			let populate = context.params.query.$populate;
+			if (typeof populate === 'undefined') {
+				populate = ['courseId', 'courseGroupId'];
+			} else if (
+				Array.isArray(populate) &&
+				!populate.includes('courseId')
+			) {
+				populate.push('courseId');
+				populate.push('courseGroupId');
 			}
-			if ('courseGroupId' in context.result) {
-				context.result.courseGroupId = context.result.courseGroupId._id;
-			} else {
-				context.result.courseId = context.result.courseId._id;
+			context.params.query.$populate = populate;
+		} else {
+			// after-hook
+			if (context.method === 'get' && (context.result || {})._id) {
+				let tempLesson = [context.result];
+				tempLesson = tempLesson.filter((lesson) => {
+					if ('courseGroupId' in lesson) {
+						return userIsInThatCourse(
+							user,
+							lesson.courseGroupId,
+							false,
+						);
+					}
+					return (
+						userIsInThatCourse(user, lesson.courseId, true) ||
+						(context.params.query.shareToken || {}) ===
+							(lesson.shareToken || {})
+					);
+				});
+				if (tempLesson.length === 0) {
+					throw new Forbidden(
+						"You don't have access to that lesson.",
+					);
+				}
+				if ('courseGroupId' in context.result) {
+					context.result.courseGroupId =
+						context.result.courseGroupId._id;
+				} else {
+					context.result.courseId = context.result.courseId._id;
+				}
 			}
-		}
 
-		if (context.method === 'find' && ((context.result || {}).data || []).length > 0) {
-			context.result.data = context.result.data.filter((lesson) => {
-				if ('courseGroupId' in lesson) {
-					return userIsInThatCourse(user, lesson.courseGroupId, false);
-				}
-				return userIsInThatCourse(user, lesson.courseId, true)
-					|| (context.params.query.shareToken || {}) === (lesson.shareToken || {});
-			});
+			if (
+				context.method === 'find' &&
```

Comment:
Codacy found an issue: ['&&' should be placed at the beginning of the line.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5994445)